controllers/k8ssandra/medusa_reconciler.go

```diff
@@ -52,10 +54,23 @@ func (r *K8ssandraClusterReconciler) ReconcileMedusa(
 		if res := r.reconcileMedusaConfigMap(ctx, remoteClient, kc, logger, namespace); res.Completed() {
 			return res
 		}
+
+		// Generate the Medusa main container
+		medusaMainContainer := medusa.GenerateMedusaMainContainer(dcConfig, medusaSpec, logger)
+		// Add the containers definitions in the podTemplateSpec
 		medusa.UpdateMedusaInitContainer(dcConfig, medusaSpec, logger)
-		medusa.UpdateMedusaMainContainer(dcConfig, medusaSpec, logger)
-		medusa.UpdateMedusaVolumes(dcConfig, medusaSpec, logger)
+		medusa.UpdateMedusaMainContainer(dcConfig, medusaMainContainer, logger)
+		// Create required volumes for Medusa
+		additionalVolumes := medusa.UpdateMedusaVolumes(dcConfig.PodTemplateSpec, medusaSpec, dcConfig.Cluster, logger)
```

Comment:
> Can we get this all consistent by doing everything (both volumes + additionalvolumes) in-place to start with. That'll be a good first step.

As much as I would want this, it's not possible without going against your other requirement of working on the pod template spec. The additional volumes aren't defined in the `podTemplateSpec`, they're under `dcConfig.StorageProperties`.

I'll rename `UpdateMedusaVolumes()` to `GenerateMedusaVolumes()` and have it return both volumes and additional volumes, then add them to the pods in an external loop. At least that behavior will be consistent for both types of volumes.

Reply:
Agreed. I don't like using both volumes and additionalvolumes but this will do for now, pending my investigation of what additionalVolumes is doing under the hood.